### DISCUSSION TITLE
Logical delete of items in an editable DataGrid

### DISCUSSION THREAD

**Discussion by naitken** [Megabit/Blazorise#2934] [Q&A] [answered]
Is there a way to implement a logical delete feature in an editable DataGrid? The model includes an IsDeleted field. Items where IsDeleted=true are filtered out of the grid but remain in the model. When the user deletes an item, IsDeleted is set to true but the item is not removed from the model.

In my specific scenario, the items are a child of a root object and the object graph is saved as a single unit of work.

**Comment by David-Moreira** [comment, marked as answer]
Hello,
With the current Datagrid features, you should be able to :   
- Apply a Datagrid filter for IsDeleted equals false 
- tell Datagrid that you manage the editing yourself by setting the UseInternalEditing to false. 
- override the RowRemoving or RowRemoved to apply your deletion logic.

**Reply by naitken** [reply]
Thanks. That worked great!
